src/React.Sample.Router.CoreMvc/Views/Home/Index.cshtml

```diff
@@ -1,15 +1,10 @@
 @using React.AspNet
 @using React.Router
 
-<!DOCTYPE html>
-<html>
-<head>
-	<title>React Router Sample</title>
-	<meta charset="utf-8" />
-</head>
-<body>
-	@Html.ReactRouterWithContext("RootComponent", new { })
+@{
+    Layout = "_Layout";
+}
+
+@Html.ReactRouter("RootComponent", new { }, preRender: x => x("var serverStyleSheet = new Styled.ServerStyleSheet();"), transformRender: (component) => $"serverStyleSheet.collectStyles({component})", postRender: x => ViewBag.ServerStyles = x("serverStyleSheet.getStyleTags()"))
```

Comment:
Hmm, I wonder if it's worth having something inbuilt for styled-components specifically, for example maybe an API like this? 
```
Html.ReactRouter("RootComponent", new { })
  .WithStyledComponents()
```

Reply:
Implemented as a chainable helper class [here](https://github.com/reactjs/React.NET/pull/589/files#diff-2a8446f860d13d661d944a659e7516c1R5)